File: service/app/schemas/agent.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class AskRequest(BaseModel):
    feed_id: str
    question: str = Field(min_length=1, max_length=2000)
    conversation_id: uuid.UUID | None = None


class MapPoint(BaseModel):
    lon: float
    lat: float
    label: str | None = None


class MapPayload(BaseModel):
    points: list[MapPoint] = Field(default_factory=list)


class ToolTraceEntry(BaseModel):
    tool_name: str
    arguments: dict
    result_summary: str


class AskResponse(BaseModel):
    conversation_id: uuid.UUID
    answer: str
    map_payload: MapPayload
    tool_trace: list[ToolTraceEntry]


class ConversationSummary(BaseModel):
    conversation_id: uuid.UUID
    feed_id: str
    created_at: datetime


class ConversationMessage(BaseModel):
    role: str
    content: str | list[dict]


class ConversationDetail(BaseModel):
    conversation_id: uuid.UUID
    feed_id: str
    created_at: datetime
    messages: list[ConversationMessage]

File: app/src/types/agent.ts
// Mirrors service/app/schemas/agent.py and service/app/schemas/auth.py by hand — no
// shared runtime between the Python backend and this TypeScript frontend, same
// hand-kept-in-sync convention as dataContract.ts mirrors manifest.schema.json.

export interface SignupRequest {
  org_name: string;
  email: string;
  password: string;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface TokenResponse {
  access_token: string;
  refresh_token: string;
  token_type: string;
}

export interface AskRequest {
  feed_id: string;
  question: string;
  conversation_id?: string | null;
}

export interface MapPoint {
  lon: number;
  lat: number;
  label?: string | null;
}

export interface MapPayload {
  points: MapPoint[];
}

export interface ToolTraceEntry {
  tool_name: string;
  arguments: Record<string, unknown>;
  result_summary: string;
}

export interface AskResponse {
  conversation_id: string;
  answer: string;
  map_payload: MapPayload;
  tool_trace: ToolTraceEntry[];
}
